nilearn/glm/first_level/first_level.py

```diff
@@ -72,6 +73,28 @@ def _ar_model_fit(X, val, Y):
     return ARModel(X, val).fit(Y)
 
 
+def _yule_walker(X, order):
+    """Compute Yule-Walker (adapted from MNE and statsmodels).
+    Operates in-place.
+    """
+    from scipy import linalg
+    if order < 1:
+        raise ValueError("AR order must be positive")
+    if type(order) is not int:
+        raise TypeError("AR order must be an integer")
+
+    denom = X.shape[-1] - np.arange(order + 1)
+    r = np.zeros(order + 1, np.float64)
+    X = X.ravel()
+    X -= X.mean()
```

Comment:
are you sure we want to have inplace operations here. It seems inplicit from the function call

Reply:
Im not 100% sure I understood your comment correctly. Did the change I make fix this?